app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js

```diff
@@ -0,0 +1,241 @@
+import { errorCodes, ethErrors } from 'eth-rpc-errors';
+import { ApprovalType } from '@metamask/controller-utils';
+
+import {
+  BUILT_IN_INFURA_NETWORKS,
+  CHAIN_ID_TO_RPC_URL_MAP,
+  CHAIN_ID_TO_TYPE_MAP,
+  CURRENCY_SYMBOLS,
+  NETWORK_TO_NAME_MAP,
+} from '../../../../../shared/constants/network';
+import {
+  isPrefixedFormattedHexString,
+  isSafeChainId,
+} from '../../../../../shared/modules/network.utils';
+import { CaveatTypes } from '../../../../../shared/constants/permissions';
+import { UNKNOWN_TICKER_SYMBOL } from '../../../../../shared/constants/app';
+import { PermissionNames } from '../../../controllers/permissions';
+import { getValidUrl } from '../../util';
+
+export function findExistingNetwork(chainId, findNetworkConfigurationBy) {
```

Comment:
Again this will change following https://github.com/MetaMask/core/pull/4286 — Infura networks will be integrated into network configurations and it will be possible to look it up by calling `getNetworkConfigurationByChainId` on the network controller. So you will be able to simplify this to:

``` typescript
export function findExistingNetwork(chainId, getNetworkConfigurationByChainId) {
  return getNetworkConfigurationByChainId(chainId);
}
```

or even just inline this function into wherever it's being used.